### PR TITLE
Add mock server for unittest

### PR DESCRIPTION
**Pull request by yhmo**
Signed-off-by: yhmo <yihua.mo@zilliz.com>

### CODE REVIEW

File: src/main/java/io/milvus/param/collection/CreateCollectionParam.java [hunk 1]
```diff
@@ -58,9 +58,9 @@ public String getDescription() {
 
 
     public static final class Builder {
-        private String collectionName;
-        private int shardsNum;
-        private String description;
+        private String collectionName = "";
```
Comment (xiaofan-luan) [inline src/main/java/io/milvus/param/collection/CreateCollectionParam.java]: The default collectionName set to "" is not a good idea, user may create a collectionName with empty string. 
Better to throw exception if collection name is null when call build

File: src/main/java/io/milvus/param/R.java [hunk 2]
```diff
@@ -60,17 +62,17 @@ public void setData(T data) {
         return r;
     }
 
-    public static <T> R<T> failed(ErrorCode errorCode) {
+    public static <T> R<T> failed(ErrorCode errorCode, String msg) {
```
Comment (xiaofan-luan) [inline src/main/java/io/milvus/param/R.java:65]: Why do we directly call failed(ErrorCode errorCode, Exception e)?

Comment (yhmo) [reply src/main/java/io/milvus/param/R.java:65]: No need to always construct an Exception object, easy to call.

File: src/main/java/io/milvus/param/collection/CreateCollectionParam.java (hunk 1 above)
Comment (yhmo) [reply src/main/java/io/milvus/param/collection/CreateCollectionParam.java]: make sense

**Review by xiaofan-luan — APPROVED** [review]

**Pull request merged**